agent/agent_test.go

```diff
@@ -852,6 +852,32 @@ func TestAgent_AddServiceWithH2PINGCheck(t *testing.T) {
 	requireCheckExists(t, a, "test-h2ping-check")
 }
 
+func TestAgent_AddServiceWithH2CPINGCheck(t *testing.T) {
+	t.Parallel()
+	a := NewTestAgent(t, "")
+	defer a.Shutdown()
+	check := []*structs.CheckType{
+		{
+			CheckID:          "test-h2cping-check",
+			Name:             "test-h2cping-check",
+			H2PING:           "localhost:12345",
+			TLSSkipVerify:    true,
```

Comment:
After some discussion internally, we came to the conclusion that it would better to reverse the sense of the option; maybe call it 'H2PingUseTLS' with a default of 'true.' This would keep parallel construction with the GRPC health check.

Happy to help on the details.